remote-sr/src/main/java/de/unistuttgart/iaas/amyassist/amy/remotesr/RemoteSR.java

```diff
@@ -0,0 +1,343 @@
+		try {
+			String file = generateTempProfile();
+
+			Properties config = this.configurationManager.getConfigurationWithDefaults(CONFIG_NAME);
+
+			String chromePath = config.getProperty(CHROME_DIRECTORY_CONFIG_KEY);
+
+			if (chromePath == null || chromePath.isEmpty()) {
+				this.logger.error("Chrome command not set. Set your chrome installation directory in {}.properties",
+						CONFIG_NAME);
+			}
+
+			this.chromeProcess = new ProcessBuilder(chromePath, "http://localhost:8080/rest/remotesr",
+					"--user-data-dir=" + file).start();
+
+			watchProcess(new BufferedReader(
+					new InputStreamReader(this.chromeProcess.getInputStream(), StandardCharsets.UTF_8)));
+		} catch (IOException e) {
+			throw new LaunchChromeException("Couldn't start Chrome:", e);
+		}
+	}
+
+	/**
+	 *
+	 * @return file path of chrome profile in temp directory
+	 * @throws LaunchChromeException
+	 *             if any error occurs
+	 */
+	private String generateTempProfile() throws LaunchChromeException {
+		try {
+			URL chromeProfileJarDir = getClass().getClassLoader().getResource("chrome_profile");
+			if (chromeProfileJarDir == null)
+				throw new LaunchChromeException("Couldn't get chrome_profile resource");
+
+			Path tempDir = Files.createTempDirectory(getClass().getSimpleName());
+
+			copyFromJar(chromeProfileJarDir.toURI(), tempDir);
+
+			return tempDir.toAbsolutePath().toString();
+
+		} catch (IOException | URISyntaxException e) {
+			throw new LaunchChromeException("Couldn't copy chrome_profile:", e);
+		}
+	}
+
+	/**
+	 * Copy a directory from our jar recursively to a directory outside of the jar
+	 * 
+	 * @param sourceURI
+	 *            URI of the source directory
+	 * @param target
+	 *            target directory path
+	 * @throws IOException
+	 *             if any IO error occurs
+	 */
+	private void copyFromJar(final URI sourceURI, final Path target) throws IOException {
+
+		FileSystem fs = null;
+		try {
+			if (sourceURI.getScheme().equals("jar")) {
+				fs = FileSystems.newFileSystem(sourceURI, Collections.<String, String> emptyMap());
+			}
+			final Path jarPath = Paths.get(sourceURI);
+
+			Files.walkFileTree(jarPath, new SimpleFileVisitor<Path>() {
+				@Override
+				public FileVisitResult preVisitDirectory(final Path dir, final BasicFileAttributes attrs)
+						throws IOException {
+					Path currentTarget = target.resolve(jarPath.relativize(dir).toString());
+					Files.createDirectories(currentTarget);
+					return FileVisitResult.CONTINUE;
+				}
+
+				@Override
+				public FileVisitResult visitFile(final Path file, final BasicFileAttributes attrs) throws IOException {
+					Files.copy(file, target.resolve(jarPath.relativize(file).toString()),
+							StandardCopyOption.REPLACE_EXISTING);
+					return FileVisitResult.CONTINUE;
+				}
+			});
+		} finally {
+			if (fs != null) {
+				fs.close();
+			}
+		}
+	}
+
+	private void watchProcess(BufferedReader reader) {
+		new Thread(() -> {
```

Comment:
give this thread a name

Reply:
fixed, GitHub doesn't detect it because it was on a different line